docs/parameters.rst

```diff
@@ -1,277 +1,179 @@
 .. _standard-parameters:
 
-Parameters
-==========
+Standard parameters
+===================
 
-OpenSCM follows a number of conventions related to parameters, we document these
-In OpenSCM a 'parameter' is anything which is passed to a model e.g. |CO2| emissions, equilibrium climate sensitivity, aerosol forcing scaling.
+In OpenSCM a ':ref:`parameter <parameter-hierarchy>`' is any named input or output variable of a mode, e.g. |CO2| emissions, equilibrium climate sensitivity, aerosol forcing scaling.
+As described :ref:`here <parameter-hierarchy>`, parameters are given in a hierarchy, e.g. ``Emissions`` -> ``CO2`` -> ``Industrial``.
 
 Simple climate models come in many different shapes and forms hence we do not expect them to all be able to do everything.
-However, to be included in OpenSCM they must be able to understand all of OpenSCM's key parameters where here, 'understand' means that their adapters should either be able to use them or should throw sensible warning/error messages if they are used.
+However, to be included in OpenSCM they should make sure their parameters fit into these standard parameters as far as possible to ensure models can be interchanged easily.
```

Comment:
I don't like this change. They have to understand all of the key parameters, where 'understand' means that their adapters should either be able to use them or should throw sensible warning/error messages if they are used. If they just blow up when given a key parameter then they're useless.

Reply:
Determining which parametres were used should not be task of an adapter (see #123).

> If they just blow up when given a key parameter then they're useless.

Why should they?

Reply:
We're not on the same page. 

What I want to make clear is that the user should always be able to pass key parameters e.g. `ECS` or `TCR` to a high-level `run` call and get sensible output i.e. things either run, a clear warning is raised or a clear error is raised. We don't want users passing 'standard' parameters in, only to receive a cryptic error message (which is possible if the implementer of an adapter doesn't think about how all (probably 4 or 5) of the key parameters would be handled). 

The text you've suggested does not make this clear ('as far as possible' is vague and doesn't tell the implementer what they actually have to do). The text I have clearly also fails to communicate what I wanted so I think we need to re-write this sentence somehow.

Doing this checking has to be done somewhere. I thought the most natural place was an adapter but maybe I'm missing something?

Reply:
> We're not on the same page.

No ;)

> What I want to make clear is that the user should always be able to pass key parameters e.g. `ECS` or `TCR` to a high-level `run` call and get sensible output i.e. things either run, a clear warning is raised or a clear error is raised. We don't want users passing 'standard' parameters in, only to receive a cryptic error message (which is possible if the implementer of an adapter doesn't think about how all (probably 4 or 5) of the key parameters would be handled).

I don't understand why that would lead to errors at all if parameters are simply ignored by the adapter because it does not know about them. I think we must not make all adapters use every parameter described here, many models would not be able to use many of these anyway (e.g. non-carbon-gases for DICE). Adapter developers should only make sure that the parameters for the respective model are consistent with these standard parameters (e.g. giving ECS a different name).

> The text you've suggested does not make this clear ('as far as possible' is vague and doesn't tell the implementer what they actually have to do). The text I have clearly also fails to communicate what I wanted so I think we need to re-write this sentence somehow.

Maybe like the text in m,y previous reply?

> Doing this checking has to be done somewhere. I thought the most natural place was an adapter but maybe I'm missing something?

I would do that in the Core so adapters do not have to be altered when we add or change standard parameters.

Reply:
Right I get it now. Handling this in core will also work so let's go for something like

"
However, to be included in OpenSCM they should make sure their parameters fit into these standard parameters as far as possible. 
"

We can then work out the details as we start building useful tools around the core <-> adapter connection